pkg/scenario/stage.go

```diff
@@ -166,6 +185,50 @@ func (c *StageClient) PlanWithSpecificVariableValueToExpect(t *testing.T, option
 	compareValues(t, actualValue, expectedValue, variable)
 }
 
+// PlanAndAssertJSONWithJSONPath performs JSON path planning and assertion in Go testing.
+//
+// t *testing.T: Testing object
+// options *terraform.Options: Terraform options
+// testCases []JSONPathTestCases: Array of JSON path test cases
+func (c *StageClient) PlanAndAssertJSONWithJSONPath(t *testing.T, options *terraform.Options, testCases []JSONPathTestCases) {
+	jsonPlan := terraform.InitAndPlanAndShow(t, options)
+
+	for _, testCase := range testCases {
+		t.Run(testCase.TestName, func(t *testing.T) {
+			var result []interface{}
+			k8s.UnmarshalJSONPath(t, []byte(jsonPlan), testCase.JSONPathToCompare, &result)
+			assert.NotNil(t, result)
+
+			// if expected is slice then it's ok to compare entire slice
+			if reflect.TypeOf(testCase.ExpectedValue).Kind() == reflect.Slice {
+				assert.ObjectsAreEqual(testCase.ExpectedValue, result)
+			} else {
+				// work on the result[0]
+				t.Logf("result returned raw: %v", result)
+				v := result[0]
+				if !testCase.AllowDifferentType {
+					assert.Equal(t, reflect.TypeOf(testCase.ExpectedValue).Kind(), reflect.TypeOf(v).Kind())
+				}
+				applyTestType(t, testCase.TestType, v, testCase.ExpectedValue, fmt.Sprintf("JSONPATH query: %s", testCase.JSONPathToCompare))
+			}
+		})
+	}
+}
+
+func applyTestType(t *testing.T, testType TestType, actual, expected interface{}, additionalMessage string) {
```

Comment:
Nice. It seems that this function can be used and even exposed as a package API, since it's quite generic. What do you think if you make it public? — Currently, it's private o this particular package.

Reply:
we can have a next feature to eternalized it, however is in the scope of this JSONPath Test Case functionality so it's not meant to be public. Yet.

Reply:
That makes sense 🍺